packages/react-relay/classic/legacy/store/__tests__/recycleNodesInto-test.js

```diff
@@ -142,6 +164,13 @@ describe('recycleNodesInto', () => {
       expect(recycleNodesInto(prevData, nextData)).toBe(prevData);
     });
 
+    it('recycles arrays with equal objects without mutating frozen `nextData`', () => {
+      const prevData = [{foo: 1}, {bar: 2}];
+      const nextData = [{foo: 1}, {bar: 2}];
+      Object.freeze(nextData);
+      expect(recycleNodesInto(prevData, nextData)).toBe(prevData);
+    });
```

Comment:
Can we add another test for recycling in objects as well?

Reply:
Sure thing!

Reply:
I went to add the requested test, but it looks like I already had one that does this:
`does not mutate frozen equal parent objects with equal leaf objects`

So this should be good to go.